modules/compiler/spirv-ll/source/builder_core.cpp

```diff
@@ -303,35 +306,35 @@ template <>
 llvm::Error Builder::create<OpExtInstImport>(const OpExtInstImport *op) {
   auto name = op->Name();
   if (name == "GLSL.std.450") {
+    ext_inst_handlers[ExtendedInstrSet::GLSL450] =
+        std::make_unique<GLSLBuilder>(*this, module);
```

Comment:
Can we assert here that we didn't have this builder before (that is, that we only encountered the `OpExtInstImport` once) by using `ext_inst_handlers.insert` and asserting on the result?

Reply:
Yeah, something like that might be good. But from what I could tell from the spec (and `spirv-val`) there's no reason a program can't import the same extension multiple times:

```
%1 = OpExtInstImport "GLSL.std.450"
%2 = OpExtInstImport "GLSL.std.450"

%3 = OpExtInst %1 Foo ...
%4 = OpExtInst %2 Bar ...
```

So I think we'd want to handle that. I was planning on just making those share a builder (hence the current structure of enum to handler as opposed to `<id> Set` to handler) as I can't see why multiple builders would need their own private data.

In reality there's kind of a two-layer map going on. From `<id> Set` to `ExtendedInstrSet` to handler.

Reply:
Oh, if that's valid, sharing the builder makes sense to me. In that case, something like
```c++
    auto &builder = ext_inst_handlers[ExtendedInstrSet::GLSL450];
    if (!builder)
        builder = std::make_unique<GLSLBUilder>(*this, module);
```
should work to avoid multiple instantiations.

Reply:
Yeah. I'll probably hide this behind a `Module` method too while I'm at it.

Reply:
Ah this currently is in `Builder`. We'll have to think later about how to make this more friendly to external users, as the `Builder` isn't exposed to users at all. Users they call `spirv_ll::Context::translate` as their entry point. Ideally we'd probably want some kind of `Context` method for registering builder handlers before calling `translate`, and that would have to set things up correctly. This would probably need a bit of re-working, and I think it can wait till later.

I'll just stick a method in `Builder` and we can move it later if we so wish.

Reply:
Done